src/binaryen-c.cpp

```diff
@@ -208,6 +209,26 @@ size_t noteExpression(BinaryenExpressionRef expression) {
   return id;
 }
 
+// We would normally use the size of `expressionRunners` as the next index, but
+// since we are going to delete runners the same address can become reused,
+// which would result in unpredictable sizes (indexes) due to undefined
+// behavior. Use a sequential id instead.
+static size_t nextExpressionRunnerId = 0;
+
+// Even though unlikely, it is possible that we are trying to use an id that is
+// still in use after wrapping around, which we must prevent.
+std::unordered_set<size_t> usedExpressionRunnerIds;
```

Comment:
This should probably be `static`, too. You could get extra fancy by making both of these helpers `static` variables inside of `noteExpressionRunner` to limit their scope, but I'll leave that up to you. OTOH, it would probably be better to just say we don't support making more than max size_t expression runners and get rid of all this logic, especially since it is literally impossible to have than many expression runners recorded in `expressionRunners`.

Reply:
Mostly thinking in terms of a very long lived process using Binaryen, let's say where modules are being created as-a-service. While we can't store max size_t in the structure, we might at some point overflow, where the likely scenario is that this is just fine, yet guarding for not reusing something left over seems like a good precaution to have. Unlikely that someone will do this with tracing enabled, ofc.

Reply:
Aha, I had missed that the `ExpressionRunner`s were removed from the `expressionRunners` map when they were destroyed 👍